cetmix_tower_server/models/cetmix_tower.py

```diff
@@ -99,3 +103,75 @@ def server_get_variable_value(
         if not value and check_global:
             value = result.get("global")
         return value
+
+    @api.model
+    def server_check_ssh_connection(self, server_reference, attempts=5, timeout=15):
+        """Check if SSH connection to the server is available.
+
+        Args:
+            server_reference (Char): Server reference.
+            attempts (int): Number of attempts to try the connection.
+                Default is 5.
+            timeout (int): Timeout in seconds for each connection attempt.
+                Default is 15 seconds.
+        Raises:
+            ValidationError:
+                If the provided server reference is invalid or
+                the server cannot be found.
+        Returns:
+            dict: {
+                "code": int,
+                    # 0 for success,
+                    # 408 if the SSH connection timed out after all attempts,
+                    # 503 if there was a generic SSH connection error.
+                "message": str  # Description of the result
+            }
+        """
+        server = self.env["cx.tower.server"].get_by_reference(server_reference)
+        if not server:
+            raise ValidationError(_("No server found for the provided reference."))
+
+        # Prepare connection details
+        ssh_host = server.ip_v4_address or server.ip_v6_address
+        ssh_port = server.ssh_port if server.ssh_port else 22
```

Comment:
This is a required [field](https://github.com/cetmix/cetmix-tower/blob/de3ef9b57e5ca302d16e6aebeec1702c011ef03d/cetmix_tower_server/models/cx_tower_server.py#L284), it always has a value.